test/run_test.py

```diff
@@ -677,6 +704,60 @@ def run_doctests(test_module, test_directory, options):
     return result
 
 
+def print_log_file(test, file_path):
+    with open(file_path, "r") as f:
+        print_to_stderr("")
+        print_to_stderr(f'PRINT LOG FILE of {test} ({file_path})')
+        print_to_stderr(f'##[group]PRINT LOG FILE of {test} ({file_path})')
+        print_to_stderr(f.read())
+        print_to_stderr('##[endgroup]')
+        print_to_stderr(f"FINISHED PRINT LOG FILE of {test} ({file_path})")
+        print_to_stderr("")
+    os.remove(file_path)
+
+
+def run_test_ops(test_module, test_directory, options):
+    if 'slow-gradcheck' in os.getenv("BUILD_ENVIRONMENT", ""):
+        # there are a lot of tests that take up a lot of space in slowgrad check, so don't bother parallelizing
+        # it's also on periodic so we don't care about TTS as much
+        return run_test(test_module, test_directory, copy.deepcopy(options),
+                        extra_unittest_args=["--use-pytest", '-vv', '-x', '--reruns=2', '-rfEX'],
+                        )
+
+    file_names = []
+    return_codes = []
+    num_procs = 3
+    os.environ["PARALLEL_TESTING"] = "1"
+    pool = mp.Pool(num_procs)
+    for i in range(num_procs):
+        log_fd, file_path = tempfile.mkstemp(dir=REPO_ROOT / "test" / "test-reports",
+                                             prefix=test_module.replace("\\", "-").replace("/", "-"))
+        return_code = pool.apply_async(run_test, args=(test_module, test_directory, copy.deepcopy(options)),
+                                       kwds={"extra_unittest_args": ["--use-pytest", '-vv', '-x', '--reruns=2', '-rfEX',
+                                                                     f'--shard-id={i}', f'--num-shards={num_procs}',
+                                                                     "-k=not _linalg_cholesky_"],
```

Comment:
`_linalg_cholesky_` is a bad one?

Reply:
yeah :( i think it's a recent addition/i didnt fail before